### PR TITLE
Add methods for reading Dietary: Energy Consumed and Dietary: Protein

### PR DESCRIPTION
**Pull request by fakeheal**
## Description

Added methods for accessing samples with information about energy and protein consumed.

------

I would like to mention that I am relatively new to react native, I have never written a single line of code in Objective-C & this is the first native method that I export to react native. I basically copied what was made for `getCarbohydratesSamples()` and changed the same files like the [PR for adding another new feature reagarding Blood Type](https://github.com/agencyenterprise/react-native-health/commit/1d0c791a8ad8acfa31be9c7680baf9609322eeb7).

I decided to put those methods where `saveFood()` is already put: `RCTAppleHealthKit+Methods_Dietary`. In Apple's documentation Energy Consumed and Protein are both in "Dietary". However the existing `getCarbohydratesSamples()` method is in `RCTAppleHealthKit+Methods_Results.m`.

Please let me know if I have to move them.

## Type of change
- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have checked my code and corrected any misspellings

-----

### CODE REVIEW

**Comment by Macelai** [comment]
Hi @fakeheal, nice to see you again here! Thanks for the contribution 😄

File: docs/getEnergyConsumedSamples.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# getEnergyConsumedSamples
+
+A quantity sample type that measures the amount of energy consumed.
+
+Permission required:
+
+- `AppleHealthKit.Constants.Permissions.EnergyConsumed`
```
Comment (Macelai) [inline docs/getEnergyConsumedSamples.md:7]: I think this permission is wrong, it should be `DietaryEnergy`. When I tested this didn't work. I don't think we have this permission on the enum. Can you check that? How did you tested this and not got permission error?

File: README.md [hunk 2]
```diff
@@ -178,6 +178,8 @@ they are splitted in the following categories
 
 ### Dietary Methods
 
+- [getEnergyConsumedSamples](/docs/getEnergyConsumedSamples.md)
```
Comment (Macelai) [inline README.md:181]: 🚀

**Review by Macelai — CHANGES_REQUESTED** [review]

File: docs/getEnergyConsumedSamples.md (hunk 1 above)
Comment (fakeheal) [reply docs/getEnergyConsumedSamples.md:7]: My bad, it seems I have created a mess. It is supposed to be `EnergyConsumed` because that's how Apple renamed the property recently. It used to be called "Dietary Energy" in both Health Apps and their SDK.

In my previous PR I added all missing permissions for Nutrition Data. However this second (`calories-and-protein-consumed`) branch I have created didn't include my changes from the previous branch. 

I am really new to this, I will have to read how to make sure I am having the latest version of the repository I have forked.

So basically, I merged my first branch `fix-read-permissions-for-nutrition` into `calories-and-protein-consumed` (the one in this PR) & everything looks and works great. (Permissions are requested and data is fetched).

How should I proceed? I can create a demo repository to test it yourself? I see that github already attached my merge of branches to this discussion.

**Review by Macelai — APPROVED** [review]

File: docs/getEnergyConsumedSamples.md (hunk 1 above)
Comment (fakeheal) [reply docs/getEnergyConsumedSamples.md:7]: @Macelai I think this one is resolved, right? :D

Comment (Macelai) [reply docs/getEnergyConsumedSamples.md:7]: About this one yeah, but looks like we have some conflicts, can you update this branch with changes required by @lucaspbordignon on the other PR and solve the conflicts, it should be easy to solve them, with you need help, please let me know!

**Review by lucaspbordignon — APPROVED** [review]

**Comment by lucaspbordignon** [comment]
@fakeheal Could you just solve the merge conflicts?

Don't want to mess with your changes and to keep the history of commits all coming from your end, as you are the main contributor of this feature 🙂
